README.md

```diff
@@ -1,34 +1,141 @@
+# Testing your code
+
+To test your code, you have two options:
+
+1. Write your own test code in `scratch.cpp`
+2. Run the tests contained in `tests.cpp`. These are the same tests that the
+   autograder will run. To choose a part to test, you need to pass the part's
+   number as a command-line argument. To do that, you'll need to open the
+   drop-down menu to the left of the green play button, open the drop-down menu
+   under "tests", click "Edit", and put the part number in the "Program
+   arguments" field. The valid options are 1, 2, 3, 4, 5, 6, and all. In
+   `tests.cpp`, there are six functions, one to test each part. They are called
+   `part1`, `part2`, etc. You can jump to the appropriate function to see the
+   code that the autograder will run on your linked list. The autograder is
+   expecting the output of `tests.cpp` to exactly match `part1_key.txt` for part
+   1, `part2_key.txt` for part 2, etc.
+
+To switch between the two methods of testing listed above, you can use the
+drop-down menu to the left of the green play button. You'll click on either
+"scratch" or "tests", depending on which one you want to use.
+
+# Implementing the lab
+
+If you find yourself confused, you might find
+[this video](https://youtu.be/zRdZaBSqjEM) helpful. Otherwise, you can work
+through the tasks below to implement your linked list class:
+
+## Part 1 (20 points) - Node class, constructor, `toString()`, and `insertHead()`
+
+Let's start off by creating a constructor and implementing `insertHead` and
+`toString` so that we can see what's going on in our linked list.
+
+Before implementing any methods, you'll need to create a `Node` class that can
+hold a value and point to the next node. After that, you'll need to add a
+private data member to the `LinkedList` for your `head` pointer. Once you've
+done that, you're ready to start implementing `LinkedList`'s methods.
+
+The constructor should be pretty simple; all it really needs to do is set `head`
+to `nullptr`.
+
+After getting the constructor working, implement the `insertHead` method. This
+method should:
+1. Check that the given value is not already in the list. If it is, `insertHead`
+   should return without adding anything to the list.
+2. Create a new node that contains the given value.
+3. Set the `next` pointer of that node to be the current `head` pointer.
+4. Change the `head` pointer to point to the newly created node.
+
+Lastly, implement `toString` so that you can see what your list looks like. The
+`toString` method should create a string that contains each element of the list
+separated by spaces. For instance, if your list contained the values 10, 5, 90,
+-12, and 7, `toString` would return `"10 5 90 -12 7"`.
+
+## Part 2 (10 points) - `size()`
+
+The next step is to implement the `size` method. This method should walk through
+your linked list, counting up the number of nodes.
```

Comment:
Let's not tell them how to implement this. They may choose to store a `size_t` as a private variable.